### PR TITLE
Fix extra output variables in ESO/SQL when monthly tabular reports ask for same

### PR DESCRIPTION
**Pull request by JasonGlazer**
Addresses #6106 

When output variables have specific key values and those output variables are also part of monthly tabular reports, all key values were being put into the ESO/SQL (and normal CSV file). This fix makes it so only the specific key values requested will show up in the ESO/SQL file.

### Review Checklist
This will not be exhaustively relevant to every PR.
 - [ ] Code style (parentheses padding, variable names)
 - [ ] Functional code review (it has to work!)
 - [ ] If defect, results of running current develop vs this branch should exhibit the fix
 - [ ] CI status: all green or justified
 - [ ] Performance: CI Linux results include performance check -- verify this
 - [ ] Unit Test(s)
 - C++ checks:
   - [ ] Argument types
   - [ ] If any virtual classes, ensure virtual destructor included, other things
 - IDD changes:
   - [ ] Verify naming conventions and styles, memos and notes and defaults
   - [ ] Open windows IDF Editor with modified IDD to check for errors
   - [ ] If transition, add rules to spreadsheet
   - [ ] If transition, add transition source
   - [ ] If transition, update idfs
 - [ ] If new idf included, locally check the err file and other outputs
 - [ ] Documentation changes in place
 - [ ] Changed docs build successfully
 - [ ] ExpandObjects changes?
 - [ ] If output changes, including tabular output structure, add to output rules file for interfaces

### CODE REVIEW

**Comment by JasonGlazer** [comment]
@mbadams5 it looks like it was just a small fix, but it took me a while to figure it out. Could you please take a look and see if you think it would break something else? 

@mjwitte thanks for the detailed suggestions on the issue.

**Comment by mbadams5** [comment]
@JasonGlazer I double checked the [original code back in 8.5.0](https://github.com/NREL/EnergyPlus/blob/c87e61b44bce627bc0fac6f0574d9b018015a3be/src/EnergyPlus/OutputProcessor.cc#L709) and your change is similar to the original functionality but with the expanded regex functionality. I don't remember why I used the `FindItemInVariableList` function there instead but this seems like the correct fix.

**Comment by JasonGlazer** [comment]
@mbadams5 and the RE2 flag of:

`(?i) `

is correct and in the right place? I have not used RE2 before, and I just wanted to verify with you that is correct,

**Comment by mbadams5** [comment]
Yes it is correct and in the right place. It is what I use for [case-insensitive search in another part of the code.](https://github.com/NREL/EnergyPlus/blob/91933e29e35a5e085d8e9ddfe8a7d9ba21939623/src/EnergyPlus/DataOutputs.cc#L121)

**Comment by JasonGlazer** [comment]
Thanks.

**Comment by JasonGlazer** [comment]
The ERR file differences were from IDF files explicitly requesting variables that were not available. These were fixed in the latest commits.

The AUD and MTD file differences are as expected and should reoccur.

**Comment by JasonGlazer** [comment]
Looked at the CI results and the look ok to me.

5ZoneAirCooledWithDOASAirLoop
- AUD (fewer IDF lines, fewer fields with defaults, fewer considered output variables) OK
- Table (fewer IDF object sin object count table, fewer fields with defaults) OK

SurfaceTest_SurfLWR
- AUD (fewer IDF lines, fewer fields with defaults, fewer considered output variables) OK
- Table (one less IDF object in object count table, fewer fields with defaults) OK

5ZoneBoilerOutsideAirReset
- AUD (fewer total and actual RVariables) [ok]
- MTD (different numbered meters) [ok]

5ZoneWaterSystems
- AUD (fewer total and actual RVariables) [ok]
- MTD (different numbered meters) [ok]

SuperMarketDetailed_DesuperHeatingCoil
- AUD (fewer total and actual RVariables) [ok]
- MTD (different numbered meters) [ok]

_SingleZoneTestCondFDTwoDD
- AUD (fewer IDF lines, fewer fields with defaults, fewer considered output variables) OK

@mjwitte or @Myoldmopar please review this, I think it is ready to be merged.

File: src/EnergyPlus/OutputProcessor.cc [hunk 1]
```diff
@@ -698,7 +698,7 @@ namespace OutputProcessor {
 		for ( Loop = MinIndx; Loop <= MaxIndx; ++Loop ) {
 			if ( ReqRepVars( Loop ).Key.empty() ) continue;
 			if ( ! SameString( ReqRepVars( Loop ).VarName, VariableName ) ) continue;
-			if ( ! DataOutputs::FindItemInVariableList( KeyedValue, VariableName ) ) continue;
+			if ( !RE2::FullMatch( KeyedValue, "(?i)" + ReqRepVars( Loop ).Key ) ) continue;
```
Comment (Myoldmopar) [inline src/EnergyPlus/OutputProcessor.cc:701]: I'll rely on @mbadams5 agreeing with this change [here](https://github.com/NREL/EnergyPlus/pull/6303#issuecomment-329193945)

File: tst/EnergyPlus/unit/OutputProcessor.unit.cc [hunk 2]
```diff
@@ -2291,6 +2292,122 @@ namespace EnergyPlus {
...
+			NumExtraVars = 0;
+			BuildKeyVarList( "LIVING1", "ZONE TOTAL INTERNAL LATENT GAIN RATE", 1, 2 );
+			EXPECT_EQ( 1, NumExtraVars );
+
+			NumExtraVars = 0;
+			BuildKeyVarList( "GARAGE", "ZONE TOTAL INTERNAL LATENT GAIN RATE", 1, 2 );
+			EXPECT_EQ( 0, NumExtraVars );
+
+
+		}
```
Comment (Myoldmopar) [inline tst/EnergyPlus/unit/OutputProcessor.unit.cc:2407]: Tests look good to me

**Review by Myoldmopar — APPROVED** [review]
Yep, this looks good.

**Pull request merged**